pkgs/test_core/lib/src/runner.dart

```diff
@@ -351,13 +350,19 @@ class Runner {
               return true;
             }
 
-            // First check if we're a match for the overridden location.
-            if (location != null) {
-              if ((line == null || location.line == line) &&
-                  (col == null || location.column == col) &&
-                  matchesUri(location.uri)) {
-                return true;
+            // First check if we're a match for the overridden location for this
+            // item or any parents.
+            var current = test as GroupEntry?;
+            while (current != null) {
+              var location = current.location;
+              if (location != null) {
```

Comment:
nit: this is a good place to use if/case

```suggestion
              if (current.location case var location?) {
```

Reply:
Oops, I usually do that and somehow forgot here - thanks!